sklearn/ensemble/_gb.py

```diff
@@ -554,6 +562,7 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             self.train_score_ = self.train_score_[:n_stages]
             if hasattr(self, "oob_improvement_"):
                 self.oob_improvement_ = self.oob_improvement_[:n_stages]
+                self.oob_scores_ = self.oob_scores_[:n_stages]
```

Comment:
Instead of updating `self.oob_score_` at each iteration, it should make more sense to update it once here.